pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -367,3 +371,33 @@ func (h *VMController) ManageOwnerOfPVCs(_ string, vm *kubevirtv1.VirtualMachine
 
 	return h.OnVMRemove(vm)
 }
+
+func (h *VMController) SetHaltIfInsufficientResourceQuota(_ string, vm *kubevirtv1.VirtualMachine) (*kubevirtv1.VirtualMachine, error) {
+	if vm == nil || vm.DeletionTimestamp != nil || vm.Spec.Template == nil {
+		return vm, nil
+	}
+
+	if err := h.vmrCalculator.CheckVMLimits(vm); err == nil {
+		return vm, nil
+	} else if !rqutils.IsInsufficientResourceError(err) {
+		return nil, err
+	}
+
+	if err := h.stopVM(vm); err != nil {
+		return nil, err
+	}
+
+	h.recorder.Eventf(vm, corev1.EventTypeWarning, "InsufficientResourceQuota", "Set runStrategy to %s", kubevirtv1.RunStrategyHalted)
+	return vm, nil
+}
```

Comment:
there is no error available as a result event description is incorrectly formatted. we can drop the error from the event message.

```
(⎈|local:migration)➜  ~ k get events | grep Insufficient
31m         Warning   InsufficientResourceQuota   virtualmachine/sample                                  Set runStrategy to Halted due to %!s(<nil>)
```